addons/sale/wizard/sale_make_invoice_advance.py

```diff
@@ -163,23 +164,59 @@ def create_invoices(self):
                 self.product_id = self.env['product.product'].create(vals)
                 self.env['ir.config_parameter'].sudo().set_param('sale.default_deposit_product_id', self.product_id.id)
 
-            sale_line_obj = self.env['sale.order.line']
             for order in sale_orders:
-                amount, name = self._get_advance_details(order)
 
                 if self.product_id.invoice_policy != 'order':
                     raise UserError(_('The product used to invoice a down payment should have an invoice policy set to "Ordered quantities". Please update your deposit product to be able to create a deposit invoice.'))
                 if self.product_id.type != 'service':
                     raise UserError(_("The product used to invoice a down payment should be of type 'Service'. Please use another product or update this product."))
-                taxes = self.product_id.taxes_id.filtered(lambda r: not order.company_id or r.company_id == order.company_id)
-                tax_ids = order.fiscal_position_id.map_tax(taxes).ids
-                analytic_tag_ids = []
-                for line in order.order_line:
-                    analytic_tag_ids = [(4, analytic_tag.id, None) for analytic_tag in line.analytic_tag_ids]
-
-                so_line_values = self._prepare_so_line(order, analytic_tag_ids, tax_ids, amount)
-                so_line = sale_line_obj.create(so_line_values)
-                self._create_invoice(order, so_line, amount)
+
+                so_lines = []
+                advance_amount = self._get_advance_amount(order)
+                group_by_taxes = defaultdict(lambda: {
+                    'lines': self.env['sale.order.line'],
+                    'sum_price_reduce_x_quantity': 0.0,
+                    'sum_price_subtotal': 0.0,
+                    'sum_price_total': 0.0,
+                })
+                if any(tax.amount_type == 'fixed' for tax in order.fiscal_position_id.map_tax(order.order_line.tax_id).flatten_taxes_hierarchy()):
+                    # no breakdown per taxes if any tax with amount type "fixed"
+                    group_by_taxes[None]['lines'] = order.order_line
+                else:
+                    for line in order.order_line:
+                        tax_ids = tuple(sorted(order.fiscal_position_id.map_tax(line.tax_id).ids))
+                        group_by_taxes[tax_ids]['lines'] += line
+                        group_by_taxes[tax_ids]['sum_price_reduce_x_quantity'] += line.product_uom_qty * line.price_reduce
```

Comment:
You probably do not want to use the price_reduce field, it's a computed field with a specified digits attribute, which means it is rounded automatically by the ORM.
It should only be used for display purposes (that's why we reapply the discount everywhere in the code instead of using this field).

Reply:
Ok, thx. Changed